tests/test_stubs.py

```diff
@@ -285,6 +285,15 @@ def test_newtype_parameter_annotation(self):
         expected = 'def test(user_id: UserId) -> None: ...'
         assert stub.render() == expected
 
+    def test_union_nonetype_annotation(self):
+        """NoneType should always be rendered as None"""
+        sig = Signature.from_callable(UpdateSignatureHelper.has_annos)
+        sig = update_signature_args(sig, {'a': Dict[str, Dict[str, Union[str, NoneType]]]}, has_self=False,
```

Comment:
This is more complex than necessary to reveal the bug; `Dict[str, Union[str, NoneType]]` is sufficient, second level of nesting in a `Dict` isn't necessary.

Reply:
Agree, but it was my actual repro :)